src/transformers/models/llama4/modeling_llama4.py

```diff
@@ -397,7 +397,7 @@ def __init__(self, config, layer_idx):
         super().__init__()
         self.hidden_size = config.hidden_size
         self.self_attn = Llama4TextAttention(config, layer_idx)
-        self.use_chunked_attention = int((layer_idx + 1) % 4 != 0)  # <=> use rope
+        self.use_chunked_attention = layer_idx not in config.no_rope_layers
```

Comment:
should be `no_rope_layers[layer_idx]` if possible! This works better!

Reply:
I remember I was confused because in the previous implementation, `no_rope_layers` contains a `0` for the NoPE layers, so the naming was reversed.